app/helpers/sessions_helper.rb

```diff
@@ -3,11 +3,13 @@
 module SessionsHelper
   def prefilled_username
     uri = Addressable::URI.parse(session["user_return_to"])
-    if uri && uri.query_values
-      uri.query_values["username"]
-    else
-      nil
-    end
+    uri.query_values["login_hint"] if uri&.query_values
```

Comment:
Looks like it was added in the first API iteration :D https://github.com/diaspora/diaspora/commit/dec10e3de92d4b24a2a9ca661f2c71af19274881

```suggestion
    uri&.query_values&.fetch("login_hint")
```